postage_stamps.py

```diff
@@ -0,0 +1,272 @@
+    truth_data = truth_gcr.get_quantities(['uniqueId', 'ra', 'dec'])
+
+    truth_table = Table(
+        data=[truth_data['uniqueId'], truth_data['ra'], truth_data['dec']],
+        names=['truth_id', 'truth_ra', 'truth_dec'],
+        dtype=[np.int64, float, float]
+    )
+
+    truth_table['truth_ra'].unit = u.degree
+    truth_table['truth_dec'].unit = u.degree
+    return truth_table
+
+
+def get_diasrc_catalog(butler, data_id):
+    """For a collection of data Ids, return a table of all DIA sources
+
+    Args:
+        butler (Butler): A data access butler
+        data_id  (dict): A dictionary of values uniquely identifying an image
+
+    Returns:
+       An astropy table
+    """
+
+    diasrc_table = Table(
+        names=['src_id', 'src_ra', 'src_dec', 'visit', 'filter',
+               'detector'],
+        dtype=[np.int64, float, float, np.int64, str, 'U10']
+    )
+
+    visit = data_id['visit']
+    filter_ = data_id['filter']
+    detector = data_id['detector']
+    for source in butler.get('deepDiff_diaSrc', dataId=data_id):
+        id_ = source['id']
+        ra = source['coord_ra']
+        dec = source['coord_dec']
+        diasrc_table.add_row([id_, ra, dec, visit, filter_, detector])
+
+    diasrc_table['src_ra'].unit = u.rad
+    diasrc_table['src_dec'].unit = u.rad
+    return diasrc_table
+
+
+def match_dataid(butler, data_id, truth_ctlg, radius):
+    """Cross match DIA sources from a given data Id with the truth catalog
+
+    Args:
+        butler     (Butler): A data access butler
+        data_id      (dict): A dictionary of values uniquely identifying an image
+        truth_ctlg  (Table): GCR truth catalog
+        radius      (float): Match radius in arc-seconds (Default: 1)
+
+    Returns:
+        Cross match results as an astropy table
+    """
+
+    source_ctlg = get_diasrc_catalog(butler, data_id)
+    source_skyc = SkyCoord(
+        ra=source_ctlg['src_ra'], dec=source_ctlg['src_dec'])
+
+    truth_skyc = SkyCoord(
+        ra=truth_ctlg['truth_ra'], dec=truth_ctlg['truth_dec'])
+
+    idx, d2d, d3d = truth_skyc.match_to_catalog_sky(source_skyc)
+
+    out_table = copy(truth_ctlg)
+    out_table.add_columns(list(source_ctlg[idx].itercols()))
+    out_table['d2d'] = d2d
+
+    # Only keep matches within radius
+    sep_constraint = d2d < radius * u.arcsec
+    return out_table[sep_constraint]
+
+
+def match_dataid_list(butler, data_id_list, truth_ctlg, radius=1):
+    """Cross match DIA sources from a list of data Ids with the truth catalog
+
+    Args:
+        butler           (Butler): A data access butler
+        data_id_list (list[dict]): A list of data identifiers
+        truth_ctlg        (Table): GCR truth catalog
+        radius            (float): Match radius in arc-seconds (Default: 1)
+
+    Returns:
+        Cross match results as an astropy table
+    """
+
+    data_id_list = copy(data_id_list)
+    first_id = data_id_list.pop()
```

Comment:
Why the copy, pop, and special first case?

Reply:
I had a loop and a half hammer and it looked like a nail. I agree that your proposed solution is simpler.